modules/node_groups/templates/userdata.sh.tpl

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash -e
+
+# Allow user supplied pre userdata code
+${pre_userdata}
+
+sed -i '/^KUBELET_EXTRA_ARGS=/a KUBELET_EXTRA_ARGS+=" ${kubelet_extra_args}"' /etc/eks/bootstrap.sh
```

Comment:
btw @ArchiFleKs I am trying to understand why your bootstrap command is missing the cluster-name argument which should be a problem according to https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh#L266-L269

see for reference https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/templates/userdata.sh.tpl#L7 or https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/launch_templates_with_managed_node_groups/templates/userdata.sh.tpl#L10

and also misses ${bootstrap_extra_args} that @sbarbat mentioned, which would be a good addition so we can supply any of the other optional options

Reply:
The issue is with the auto merge of official AWS usersata and the custom supplied. So there is no possibility to add bootstrap args before.